### PR TITLE
Add support symfony 5

### PR DESCRIPTION
**Pull request by franmomu**
## Subject

I removed some test that looks like were added in https://github.com/sonata-project/SonataCoreBundle/pull/324 for Symfony 3.x and FQCN compatibilities.

I am targeting this branch, because these changes respect BC.

## Changelog

```markdown
### Added
- Added Support for Symfony 5 packages
```

### CODE REVIEW

**Comment by core23** [comment]
Can you check the phpstan error @franmomu ?

This could help
https://github.com/sonata-project/form-extensions/pull/52/files#diff-5d8c276c617696f9eb6923af5b12a903

**Comment by franmomu** [comment]
> Can you check the phpstan error @franmomu ?
> 
> This could help
> https://github.com/sonata-project/form-extensions/pull/52/files#diff-5d8c276c617696f9eb6923af5b12a903

The thing is that [IntlDateFormatter from Symfony doesn't implement formObject](https://github.com/symfony/symfony/blob/4.3/src/Symfony/Component/Intl/DateFormatter/IntlDateFormatter.php#L238).

But there is a problem I've just realised that I'm trying to figure it out with Symfony 5. [BaseDatePickerType](https://github.com/sonata-project/form-extensions/blob/1.x/src/Type/BasePickerType.php) has `Symfony\Component\Translation\TranslatorInterface` as an argument, in Symfony 5 `Symfony\Contracts\Translation\TranslatorInterface` should be used instead, but the problem is that this interface doesn't have the `getLocale` method, so it should be fetched from the request.

**Comment by franmomu** [comment]
So what I've done:
Move the StubTranslator and bump to php 7.2 [like CoreBundle](https://github.com/sonata-project/SonataCoreBundle/pull/723).

With `BasePickerType`, because there are two Interfaces to support, I removed the type hinting and add an internal check. The thing about these interfaces is that the current `Symfony\Component\Translation\TranslatorInterface` has a `getLocale` method and `Symfony\Contracts\Translation\TranslatorInterface` doesn't, so I added the third optional parameter [RequestStack to get the locale from there](https://github.com/symfony/symfony/issues/29530#issuecomment-445470461).

This check is to make sure that if you use the new interface (without `getLocale`), you should pass the RequestStack:

```php
if ($translator instanceof TranslatorInterface && null === $requestStack) {
    throw new \InvalidArgumentException(sprintf(
        'Argument 3 should be an instance of %s',
        RequestStack::class
    ));
}
```

I'll take a look at phpstan

File: tests/Type/DatePickerTypeTest.php [hunk 1]
```diff
@@ -85,13 +63,37 @@ public function testSubmitValidData(): void
...
 
         return [
             new PreloadedExtension([$type], []),
         ];
     }
+
+    /**
+     * @return MockObject|TranslatorInterface|LegacyTranslatorInterface\
+     */
+    private function getTranslatorMock()
```
Comment (phansys) [inline tests/Type/DatePickerTypeTest.php]:
```suggestion
    private function getTranslatorMock(): MockObject
```

File: tests/Type/DateTimePickerTypeTest.php [hunk 2]
```diff
@@ -105,13 +84,41 @@ public function testSubmitMatchingDateFormat(): void
...
 
         return [
             new PreloadedExtension([$type], []),
         ];
     }
+
+    /**
+     * @return MockObject|TranslatorInterface|LegacyTranslatorInterface\
+     */
+    private function getTranslatorMock()
```
Comment (phansys) [inline tests/Type/DateTimePickerTypeTest.php]:
```suggestion
    private function getTranslatorMock(): MockObject
```

**Comment by greg0ire** [comment]
Thanks @franmomu !

**Pull request merged**